source/dub/packagesuppliers/maven.d

```diff
@@ -114,18 +116,19 @@ class MavenRegistryPackageSupplier : PackageSupplier {
 		};
 		xml.parse();
 
-		m_metadataCache[packageId] = CacheEntry(json, now);
-		return json;
+		auto entry = CacheEntry(Metadata.fromJson(json), now);
+		m_metadataCache[packageId] = entry;
+		return typeof(return)(entry.data);
 	}
 
 	SearchResult[] searchPackages(string query)
 	{
 		// Only exact search is supported
 		// This enables retrival of dub packages on dub run
 		auto md = getMetadata(query);
-		if (md.type == Json.Type.null_)
+		auto recipe = getBestPackage(md, query, Dependency.any, true);
```

Comment:
why remove the check if metadata is null? logically would want to have it everywhere before testing getBestPackage, maybe we can merge the two functions together?

Reply:
nvm it's done in the getBestPackage function, however there are a few other cases in the code where it's still checked against before getBestPackage.